src/subdomains/supporting/payin/strategies/register/impl/base/evm.strategy.ts

```diff
@@ -165,4 +174,68 @@ export abstract class EvmStrategy extends RegisterStrategy {
 
     await this.createPayInsAndSave(newEntries, log);
   }
+
+  protected processAddressWebhookMessageQueue(dto: AlchemyWebhookDto): void {
+    this.addressWebhookMessageQueue
+      .handle<void>(async () => this.processWebhookTransactions(dto))
+      .catch((e) => {
+        this.logger.error('Error while process new payin entries', e);
+      });
+  }
+
+  private async processWebhookTransactions(dto: AlchemyWebhookDto): Promise<void> {
+    const fromAddresses = this.getOwnAddresses();
+    const toAdresses = await this.getPayInAddresses();
+
+    const relevantTransactions = this.filterWebhookTransactionsByRelevantAddresses(fromAddresses, toAdresses, dto);
+
+    const payInEntries = await this.mapWebhookTransactions(relevantTransactions);
+
+    await this.addReferenceAmounts(payInEntries);
+
+    const log = this.createNewLogObject();
+    await this.createPayInsAndSave(payInEntries, log);
+  }
+
+  private filterWebhookTransactionsByRelevantAddresses(
+    fromAddresses: string[],
+    toAddresses: string[],
+    dto: AlchemyWebhookDto,
+  ): AlchemyWebhookActivityDto[] {
+    const notFromOwnAddresses = dto.event.activity.filter(
+      (tx) => !fromAddresses.map((a) => a.toLowerCase()).includes(tx.fromAddress.toLowerCase()),
+    );
+
+    return notFromOwnAddresses.filter((tx) =>
+      toAddresses.map((a) => a.toLowerCase()).includes(tx.toAddress.toLowerCase()),
+    );
```

Comment:
Is this separated on purpose? Could be merged.

Reply:
It's the same in line 134-138. In my opinion it is helpful in this case to have the variable "notFromOwnAddresses" as a kind of documentation, what happens here.